proxy/healthy_endpoints.go

```diff
@@ -0,0 +1,34 @@
+package proxy
+
+import (
+	"math/rand"
+
+	"github.com/zalando/skipper/routing"
+)
+
+type healthyEndpoints struct {
+	rnd              *rand.Rand
+	endpointRegistry *routing.EndpointRegistry
+}
+
+func (h *healthyEndpoints) filterHealthyEndpoints(endpoints []routing.LBEndpoint, rt *routing.Route) []routing.LBEndpoint {
+	if h == nil {
+		return endpoints
+	}
```

Comment:
I don't understand this condition, if h is nil `h.filterHealthyEndpoints()` call should fail before even going into function, right?

Reply:
@MustafaSaber test it yourself and you will see :)

Reply:
note for myself, check nil in Go https://go.dev/play/p/8rZ9UOcEY7t